### DISCUSSION TITLE
Use Embed to script a table using a data provider

### DISCUSSION THREAD

**Discussion by chibinet** [Budibase/budibase#13933] [Help] [answered]
Can I write a table in CSS to be run as an embed that references a Data Provider in budibase?

Seems like a great work around for limited styling of tables if I can use the Data Provider as the source data in the scripted table.

**Comment by ConorWebb96** [comment]
Hey @chibinet,

I can't see why you wouldn't be able to. You would need to use the `embed component`. To make things easier as well you'd likely have to use JS to handle the data, create the html and then loop through the data provider's array to generate your rows to output a table with your styling.

This isn't something we would normally recommend as you won't be able to use any of our actions with such a component. So if you are happy enough to create something which is purely visual and can't be interacted with easily then by all means go ahead.

I hope this helps.

**Reply by chibinet** [reply]
Yes, COMPLETELY a visual aid. 
Not looking for any specific interactivity.

Possible you have any sample scripts that do work. Have hit a wall.

**Reply by ConorWebb96** [reply, marked as answer]
Hey @chibinet,

Yup you can definitely do this within Budibase. Here some screenshots showcasing the setup and the outcome.

<img width="1194" alt="Screenshot 2024-06-20 at 07 56 17" src="https://github.com/Budibase/budibase/assets/126772285/0aefe377-cdb1-46ed-844f-924c75e3eeff">

<img width="1199" alt="Screenshot 2024-06-20 at 07 56 58" src="https://github.com/Budibase/budibase/assets/126772285/eb784328-a230-47bf-83dc-5b21c2374052">

<img width="1512" alt="Screenshot 2024-06-20 at 07 57 32" src="https://github.com/Budibase/budibase/assets/126772285/49c51216-4190-4fdd-b6d9-d9540b800297">

I hope this helps.

**Reply by chibinet** [reply]
Perfect, Was able to take that and modify it for my use case.

found that I needed to add the Style to the JavaScript to make it work properly.

Thank you!
